File: flutter_inappwebview_platform_interface/lib/src/types/save_file_security_check_starting_request.dart
import 'package:flutter_inappwebview_internal_annotations/flutter_inappwebview_internal_annotations.dart';

import '../web_uri.dart';
import 'enum_method.dart';

part 'save_file_security_check_starting_request.g.dart';

///Class that represents the request used by the [PlatformWebViewCreationParams.onSaveFileSecurityCheckStarting] event.
@ExchangeableObject()
class SaveFileSecurityCheckStartingRequest_ {
  ///The document origin URI of the file save operation.
  WebUri? documentOriginUri;

  ///The file extension to be saved.
  String? fileExtension;

  ///The full file path of the file to be saved.
  String? filePath;

  ///Whether to cancel the save operation.
  bool? cancelSave;

  ///Whether to suppress the default policy check.
  bool? suppressDefaultPolicy;

  SaveFileSecurityCheckStartingRequest_({
    this.documentOriginUri,
    this.fileExtension,
    this.filePath,
    this.cancelSave,
    this.suppressDefaultPolicy,
  });
}

File: flutter_inappwebview_windows/windows/types/save_file_security_check_starting_request.cpp
#include "save_file_security_check_starting_request.h"

namespace flutter_inappwebview_plugin
{
  SaveFileSecurityCheckStartingRequest::SaveFileSecurityCheckStartingRequest(const std::optional<std::string>& documentOriginUri,
    const std::optional<std::string>& fileExtension,
    const std::optional<std::string>& filePath,
    const std::optional<bool>& cancelSave,
    const std::optional<bool>& suppressDefaultPolicy)
    : documentOriginUri(documentOriginUri), fileExtension(fileExtension), filePath(filePath),
    cancelSave(cancelSave), suppressDefaultPolicy(suppressDefaultPolicy)
  {}

  SaveFileSecurityCheckStartingRequest::SaveFileSecurityCheckStartingRequest(const flutter::EncodableMap& map)
    : SaveFileSecurityCheckStartingRequest(get_optional_fl_map_value<std::string>(map, "documentOriginUri"),
      get_optional_fl_map_value<std::string>(map, "fileExtension"),
      get_optional_fl_map_value<std::string>(map, "filePath"),
      get_optional_fl_map_value<bool>(map, "cancelSave"),
      get_optional_fl_map_value<bool>(map, "suppressDefaultPolicy"))
  {}

  flutter::EncodableMap SaveFileSecurityCheckStartingRequest::toEncodableMap() const
  {
    return flutter::EncodableMap{
      {"documentOriginUri", make_fl_value(documentOriginUri)},
      {"fileExtension", make_fl_value(fileExtension)},
      {"filePath", make_fl_value(filePath)},
      {"cancelSave", make_fl_value(cancelSave)},
      {"suppressDefaultPolicy", make_fl_value(suppressDefaultPolicy)}
    };
  }
}